packages/asc-ui/src/components/Toggle/Toggle.tsx

```diff
@@ -1,160 +1,37 @@
-  as?: ElementType
-} & ToggleStyleProps &
-  ToggleHandlerProps &
-  BackDropProps
-
-// Todo: refactor this to Collapse component https://github.com/Amsterdam/amsterdam-styled-components/issues/379
-const Toggle: FunctionComponent<Props & HTMLAttributes<HTMLDivElement>> = ({
-  children: childrenProps,
-  onClick,
-  open: openProp,
-  onKeyDown,
-  onOpen,
-  css,
-  render,
-  iconOpen,
-  iconClose,
-  ToggleHandler,
-  title,
-  hasBackDrop,
-  backdropOpacity,
-  zIndexOffset,
-  ariaLabel,
-  ...otherProps
-}) => {
-  const [open, setOpen] = useState(false)
-
-  const handleOnOpen = useCallback(
-    (openParam: boolean) => {
-      setOpen(openParam)
-
-      if (onOpen) {
-        onOpen(openParam)
-      }
-    },
-    // Empty dependency array prevents an infinite loop if the parent rerenders
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
-  )
-
-  const { onKeyDown: onKeyDownHook } = useActionOnEscape(() =>
-    handleOnOpen(false),
-  )
-  const ref = useRef<HTMLDivElement>(null!)
-
-  const handleOnBlur = () => {
-    setTimeout(() => {
-      const element = ref.current
-      if (element) {
-        const currentFocus = ownerDocument(element).activeElement
-        if (!element.contains(currentFocus)) {
-          handleOnOpen(false)
-        }
-      }
-    })
-  }
-
-  const handleOnKeyDown = (e: KeyboardEvent<HTMLElement>) => {
-    onKeyDownHook(e)
-    if (onKeyDown) {
-      onKeyDown(e)
-    }
-  }
-
-  const handleOnClick = (e: MouseEvent<HTMLElement>) => {
-    handleOnOpen(!open)
-
-    if (onClick) {
-      onClick(e)
-    }
-  }
-
-  const { children } = usePassPropsToChildren(childrenProps, {
-    'aria-hidden': !open,
-  })
-
-  // Useful if parent needs to take over control the isOpen state
-  useEffect(() => {
-    if (typeof openProp !== 'undefined') {
-      handleOnOpen(openProp)
-    }
-  }, [handleOnOpen, openProp])
-
-  const conditionalRenderedChildren = open ? children : null
+  useImperativeHandle(externalRef, () => ref.current as HTMLInputElement)
 
   return (
-    <ToggleStyle
-      ref={ref}
-      css={css}
-      onBlur={handleOnBlur}
-      onKeyDown={handleOnKeyDown}
-      hasBackDrop={hasBackDrop}
-      {...otherProps}
-      tabIndex={-1} // This will enable the onblur event for this div element on all browsers
-    >
-      {ToggleHandler && (
-        <ToggleHandler
-          {...{
-            open,
-            iconClose,
-            iconOpen,
-            onClick: handleOnClick,
-            title,
-            hasBackDrop,
-          }}
-          {...(ariaLabel ? { 'aria-label': ariaLabel } : {})}
-          type="button"
-        />
-      )}
-      {render ? children : conditionalRenderedChildren}
-      {hasBackDrop && open && (
-        <BackDrop
-          onClick={handleOnClick}
-          zIndexOffset={zIndexOffset}
-          backdropOpacity={backdropOpacity}
-        />
-      )}
-    </ToggleStyle>
+    <WrapperStyle className={className}>
```

Comment:
The MDN page about the Switch Role has an example of implementation like:

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Switch_role

```
<button type="button" role="switch" aria-checked="true"
    id="speakerPower" class="switch">
    <span>off</span>
    <span>on</span>
</button>
<label for="speakerPower" class="switch">Speaker power</label>
```

It seems like you have implemented a component with 2 switches with the same name. I think it's preferable to implement it with the button component and use it like a boolean control.

Something like:

```
<ToggleButton onToggle={(event, state: 'on'|'off') => { ... }} />
```

Reply:
wouldn't it be better to keep de radio-buttons because it is syntactically better for the two state toggle buttons. that is what I did for the Switch component (checkbox).

Reply:
maybe remove de role="switch" from the code

Reply:
The Toggle is much like the switch. The only thing that's different is that you provide a fixed set of optional values for either state instead of using boolean state. 

My point being more that if you want to use`role="switch"`, the implementation should be different. Semantically it reads like:

Switch 1: value A [checked/unchecked] 
Switch 2: value B [checked/unchecked]

Instead of:

Switch: value [checked/unchecked]

Reply:
Yes but you can only have 1 value: you can never switch both switches to true.  the switches are not separated from each other. 

 That is why it should be a radio button instead. don't you agree?

Reply:
No, I don't think radios are the best fit for this kind of switch/toggle/whatever-construct-with-2fixed-choices. You can remove the role="switch" if you want to continue with the radios. Radio's have their own aria semantics so no need for mixing those.

Reply:
I do not agree but I will do it anyway

Reply:
Then you shouldn't. Maybe find a second opinion?

Reply:
goed idee: ik vraag jon even

Reply:
Yeah, I think I have to agree with @timvanoostrom here. We want to have a single element that is responsible for the `switch` role. Looking at examples on the web the most common way to do this seems to be to add the `switch` role to a `button` element, and setting `aria-checked` to the right value when it is pressed.

I think this makes the most sense as well, so I would suggest we follow the implementation as specified on [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Switch_role).

Reply:
ok I will do that!